### PR TITLE
[Blaze] Fixes + unit tests for the payment viewmodels

### PR DESCRIPTION
**Pull request by hichamboushaba**
### Description
This PR has some fixes and adds unit tests of the payment viewmodels.

### Testing instructions
Green CI is enough.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentMethodsListViewModel.kt [hunk 1]
```diff
@@ -55,10 +55,11 @@ class BlazeCampaignPaymentMethodsListViewModel @Inject constructor(
         userAgent = userAgent,
         wpComWebViewAuthenticator = wpComWebViewAuthenticator,
         onUrlLoaded = { url ->
-            if (url.startsWith(navArgs.paymentMethodsData.addPaymentMethodUrls.successUrl)) {
+            val urls = navArgs.paymentMethodsData.addPaymentMethodUrls
+            if (url.startsWith(urls.successUrl)) {
                 runCatching {
                     URL(url).query?.split("&")
-                        ?.firstOrNull { it.startsWith("payment_method_id=") }
+                        ?.firstOrNull { it.startsWith("${urls.idUrlParameter}=") }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentMethodsListViewModel.kt:67]: Unit tests helped identifying this bug 😌

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryScreen.kt [hunk 2]
```diff
@@ -286,8 +285,7 @@ private fun PaymentTotals(
             )
 
             Text(
-                // TODO format correctly
-                text = "${budget.currencyCode}${budget.totalBudget}",
+                text = budget,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/payment/BlazeCampaignPaymentSummaryScreen.kt:288]: I left this TODO, but almost forgot about it.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by JorgeMucientes — APPROVED** [review]

**Comment by hichamboushaba** [comment]
@0nko I cleared your assignment because this was already approved, I just had to fix unit tests after the merge conflicts.

**Pull request merged**